### PR TITLE
creates parserfactory

### PR DESCRIPTION
**Pull request by Harsh1-1**

### CODE REVIEW

File: src/core/parser/parser.py [hunk 1]
```diff
@@ -21,98 +17,87 @@ def __init__(self, **kwargs):
...
-        raise ValueError(language)
-
-# Driver Code to test parser
-# def parser_test():
-#     filename = "./parser.py"
-#     parser_obj = Parser()
-#     print(parser_obj.getTODOPython(filename))
-
-# parser_test()
+class pythonTodoParser:
```
Comment (akshatgit) [inline src/core/parser/parser.py]: Split the class into a separate file and start the name with a capital letter.

File: src/core/parser/parser.py [hunk 2]
```diff
@@ -21,98 +17,87 @@ def __init__(self, **kwargs):
...
-                if parsed_todo_content is not None:
-                    self.parsed_todo[language].append(parsed_todo_content)
+                todos_of_file = parser.parse(filepath) # TODO Optimize this line
+                if todos_of_file is not None:
+                    self.todo_list += todos_of_file
+        return self.todo_list
+    
 
-        return self.parsed_todo
+class parserFactory:
```
Comment (akshatgit) [inline src/core/parser/parser.py]: Split the class into a separate file and start the name with a capital letter.

File: src/core/parser/parser.py (hunk 1 above)
Comment (Harsh1-1) [reply src/core/parser/parser.py]: Okay

File: src/core/parser/.gitignore [hunk 3]
```diff
@@ -0,0 +1,2 @@
+__pycache__/
+*.pyc
```
Comment (akshatgit) [inline src/core/parser/.gitignore]: Not required.. i have added these in global .gitignore
*.py[cod]
*$py.class

File: src/core/parser/language_parsers/html.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
+from src.core.util.todo import Todo
+
+class HtmlTodoParser:
```
Comment (akshatgit) [inline src/core/parser/language_parsers/html.py]: HTML instead of Html :P

**Review by akshatgit — APPROVED** [review]

**Comment by akshatgit** [comment]
LGTM

**Pull request merged**